### DISCUSSION TITLE
[@types/minimatch] is missing Minimatch class export

### DISCUSSION THREAD

**Discussion by evert** [DefinitelyTyped/DefinitelyTyped#60307] [Issues with a @types package] [answered]
The [minimatch][1] library has an exported [Minimatch][2] class (see [readme][3]).

Currently the types only describe the few functions that this library has, but using the class enables us to cache the generated regular expression.

[1]: https://github.com/isaacs/minimatch/
[2]: https://github.com/isaacs/minimatch/blob/main/minimatch.js#L901
[3]: https://github.com/isaacs/minimatch/#minimatch-class

[a comment by an automated account is not shown]

**Comment by BendingBender** [comment]
I don't understand your problem, the class is exposed, you just need to use the export: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/c1bf1945c4eab17ac4fee075ad9885223961242f/types/minimatch/index.d.ts#L30

**Comment by evert** [comment]
At the time I was not able to:

```typescript
import { Minimatch } from 'minimatch';

const minimatch = new Minimatch();
```

It's possible this has changed, but the source suggests that this is not true.

**Reply by BendingBender** [reply]
Not sure what you're trying to say. Does ist work or not? Just looked it up, the last time the definitions changed was 13 months ago. And the current tests suggest that it should work: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/8a2aab1d6bb4032875c61450ed8bda7ca0cac6e4/types/minimatch/minimatch-tests.ts#L7

**Comment by evert** [comment, marked as answer]
I'm no longer using this library, so if you think it's fixed I'll just close!
